elastic/kibana
# [Lens] Fix bug causing bar charts to render bars at the wrong width

Fixes https://github.com/elastic/kibana/issues/81580

Xy charts used to replicate the auto interval logic from the data plugin to determine the interval of the x axis datapoints. We know have an official and supported way to pull this information out of the column meta data.

This PR removes the existing logic of calculating the interval based on knowledge of internal meta data and the targeted number of buckets with a call to `data.search.aggs.getDateMetaByDatatableColumn`

As this method is async, I had to move the invocation outside of the react component which required some minor refactoring. The resulting min interval is in turn passed into the renderer as additional prop.

## Testing

To reproduce the bug, create an index with sparse data:
```
POST sparsetest/_doc
{
  "ts": "2020-11-05T00:00:00.000Z",
  "category": "A"
}

POST sparsetest/_doc
{
  "ts": "2020-11-05T03:00:00.000Z",
  "category": "A"
}

POST sparsetest/_doc
{
  "ts": "2020-11-05T05:00:00.000Z",
  "category": "A"
}

POST sparsetest/_doc
{
  "ts": "2020-11-05T09:00:00.000Z",
  "category": "A"
}
```

Then, follow these steps:
* Create index pattern
* Create Lens vis
* Put ts date histogram an x axis, category terms on "break down by"
* Set time range to `time:(from:'2020-11-04T23:36:46.504Z',to:'2020-11-05T09:56:56.456Z')`

Observe wide bars (10m interval, but 2h bars):
<img width="1452" alt="Screenshot 2020-11-17 at 15 39 15" src="https://user-images.githubusercontent.com/1508364/99403839-34037700-28eb-11eb-8910-81ccd9d67319.png">

On this PR:
<img width="1457" alt="Screenshot 2020-11-17 at 15 39 20" src="https://user-images.githubusercontent.com/1508364/99403862-3c5bb200-28eb-11eb-8594-dcda2468aa77.png">
